ejb3/src/main/java/org/jboss/as/ejb3/subsystem/deployment/TimerResourceDefinition.java

```diff
@@ -271,6 +283,9 @@ protected void readAttribute(TimerImpl timer, ModelNode toSet) {
 
             @Override
             protected void readAttribute(TimerImpl timer, ModelNode toSet) {
+                if (timer.isCanceled()) {
+                    return;
+                }
```

Comment:
This one should be removed, as the result for 'active' should be 'false' for a cancelled timer. The timer.isActive() call should not fail.